# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/script/ScriptLoadRequest.h
# Commit: 0a3d9301244b
# Full Hash: 0a3d9301244b91559c4c19314ce991525d9f605c
# Author: Yulia Startsev <ystartsev@mozilla.com>
# Date: 2022-02-18 21:52:29
# Regressor Bug: 1742437
# File Overlap Count: 1
# Description:
#   Bug 1742437 - move mElement from ScriptFetchOptions to DOMScriptLoadContext; r=jonco,smaug
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D136742
# ==============================================================================

diff -r 46d45c7bae2f -r 0a3d9301244b dom/script/ScriptLoadRequest.h
--- a/dom/script/ScriptLoadRequest.h	Fri Feb 18 12:45:48 2022 +0000
+++ b/dom/script/ScriptLoadRequest.h	Fri Feb 18 12:45:49 2022 +0000
@@ -59,7 +59,7 @@
   NS_DECL_CYCLE_COLLECTION_NATIVE_CLASS(ScriptFetchOptions)
 
   ScriptFetchOptions(mozilla::CORSMode aCORSMode,
-                     enum ReferrerPolicy aReferrerPolicy, Element* aElement,
+                     enum ReferrerPolicy aReferrerPolicy,
                      nsIPrincipal* aTriggeringPrincipal,
                      nsIGlobalObject* aWebExtGlobal);
 
@@ -341,7 +341,7 @@
   virtual ~DOMScriptLoadContext();
 
  public:
-  explicit DOMScriptLoadContext(ScriptLoadRequest* aRequest);
+  explicit DOMScriptLoadContext(Element* aElement, ScriptLoadRequest* aRequest);
 
   NS_DECL_CYCLE_COLLECTING_ISUPPORTS
   NS_DECL_CYCLE_COLLECTION_SCRIPT_HOLDER_CLASS(DOMScriptLoadContext)
@@ -456,6 +456,9 @@
   Atomic<Runnable*> mRunnable;  // Runnable created when dispatching off thread
                                 // compile. Tracked here so that it can be
                                 // properly released during cancellation.
+
+  nsCOMPtr<Element> mElement;
+
   RefPtr<ScriptLoadRequest> mRequest;
 
   // Non-null if there is a document that this request is blocking from loading.